test_conformance/spirv_new/test_spirv_15.cpp

```diff
@@ -0,0 +1,160 @@
+
+    clMemWrapper dst_ulong =
+        clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+                       sizeof(result_ulong), &result_ulong, &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create dst_ulong buffer");
+
+    clMemWrapper dst_uint2 =
+        clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+                       sizeof(result_uint2), &result_uint2, &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create dst_uint2 buffer");
+
+    error |= clSetKernelArg(kernel, 0, sizeof(dst_ulong), &dst_ulong);
+    error |= clSetKernelArg(kernel, 1, sizeof(dst_uint2), &dst_uint2);
+    SPIRV_CHECK_ERROR(error, "Failed to set kernel args");
+
+    size_t global = 1;
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &global, NULL, 0,
+                                   NULL, NULL);
+    SPIRV_CHECK_ERROR(error, "Failed to enqueue kernel");
+
+    error =
+        clEnqueueReadBuffer(queue, dst_ulong, CL_TRUE, 0, sizeof(result_ulong),
+                            &result_ulong, 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(error, "Unable to read dst_ulong buffer");
+
+    error =
+        clEnqueueReadBuffer(queue, dst_uint2, CL_TRUE, 0, sizeof(result_uint2),
+                            &result_uint2, 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(error, "Unable to read dst_uint2 buffer");
+
+    if (result_ulong != result_uint2)
+    {
+        log_error("Results mismatch!  ulong = 0x%016" PRIx64
+                  " vs. uint2 = 0x%016" PRIx64 "\n",
+                  result_ulong, result_uint2);
+        return TEST_FAIL;
+    }
+
+    return TEST_PASS;
+}
+
+REGISTER_TEST(spirv15_non_uniform_broadcast)
+{
+    if (!is_spirv_version_supported(device, "SPIR-V_1.5"))
+    {
+        log_info("SPIR-V 1.5 not supported; skipping tests.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    if (!is_extension_available(device, "cl_khr_subgroup_ballot"))
+    {
+        log_info("cl_khr_subgroup_ballot is not supported; skipping tests.\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    cl_int error = CL_SUCCESS;
+
+    clProgramWrapper prog;
+    error = get_program_with_il(prog, device, context,
+                                "spv1.5/non_uniform_broadcast_dynamic_index");
+    SPIRV_CHECK_ERROR(error, "Failed to compile spv program");
+
+    clKernelWrapper kernel = clCreateKernel(
+        prog, "non_uniform_broadcast_dynamic_index_test", &error);
+    SPIRV_CHECK_ERROR(error, "Failed to create spv kernel");
+
+    // Get the local work-group size for one sub-group per work-group.
+    size_t lws = 0;
+    size_t one = 1;
+    error = clGetKernelSubGroupInfo(
+        kernel, device, CL_KERNEL_LOCAL_SIZE_FOR_SUB_GROUP_COUNT,
+        sizeof(size_t), &one, sizeof(size_t), &lws, NULL);
+    SPIRV_CHECK_ERROR(error, "Failed to get local work size for one sub-group");
+
+    // Use four work-groups, unless the local-group size is less than four.
+    size_t wgcount = std::min<size_t>(lws, 4);
```

Comment:
I would suggest `#include <algorithm>` for `std::min`.

Reply:
done.